PyStemmusScope/utils.py

```diff
@@ -69,6 +72,103 @@ def to_absolute_path(
     return pathlike.expanduser().resolve(strict=must_exist)
 
 
+def get_forcing_file(config):
+    """Get forcing file from the location.
+    """
+    location, fmt = check_location_fmt(config["Location"])
+    # check if the forcing file exists for the given locaiton(s)
+    if fmt == "site":
+        # get forcing file list
+        forcing_filenames_list = [file.name for file in Path(config["ForcingPath"]).iterdir()]
+        forcing_file = [filename for filename in forcing_filenames_list if location in filename]
+        if not forcing_file:
+            raise ValueError(f"Forcing file does not exist for the given site {location}.")
+        if len(forcing_file) > 1:
+            raise ValueError(f"Multiple forcing files exist for the given site {location}." +
+                "Please check your focing files and remove the redundant files.")
+        forcing_file = forcing_file[0]
+
+    elif fmt == "latlon":
+        raise NotImplementedError
+    elif fmt == "bbox":
+        raise NotImplementedError
+
+    return forcing_file
+
+
+def check_location_fmt(loc):
+    """Check the format of the model location.
+
+    Three types of format are supported:
+    - Site name, e.g., "DE-Kli"
+    - Latitude and longitude, e.g. "(56.4, 112.0)"
+    - A rectangular bounding box, described with two opposing corners:
+        [[lat1, lon1], [lat2, lon2]], e.g., "[[19.5, 125.5], [20.5, 130.0]]".
```

Comment:
The example of bbox here shows nested lists e.g.`[[lat1, lon1], [lat2, lon2]]` and `bbox_pattern` looks for this pattern. However, the `Return` doctrsing tells the bbox is a `tuple of tuples`. Let's make them consistent.

Reply:
I wrote it this way, as the square brackets nicely fit interval notation. However, we'd want the bounding box to be tuples because their are immutable.